fl4health/feature_alignment/tab_features_info_encoder.py

```diff
@@ -0,0 +1,194 @@
+        self.target_info = target_info
+        self.vocabulary = vocabulary
+        self.default_fill_values = default_fill_values
+
+    def features_by_type(self, feature_type: str) -> List[str]:
+        return sorted([feature for feature, t in self.features_to_types.items() if t == feature_type])
+
+    def type_to_features(self) -> Dict[str, List[str]]:
+        return {feature_type: self.features_by_type(feature_type) for feature_type in FEATURE_TYPES}
+
+    def get_categories(self) -> Dict[str, List[Scalar]]:
+        return self.categories
+
+    def get_categories_list(self) -> List[List[Scalar]]:
+        return [self.get_categories()[feature_name] for feature_name in self.features_by_type(ORDINAL)]
+
+    def get_target(self) -> str:
+        return self.target_info.get_target()
+
+    def get_target_type(self) -> str:
+        return self.target_info.get_target_type()
+
+    def get_target_categories(self) -> List[Scalar]:
+        return self.target_info.get_target_categories()
+
+    def get_vocabulary(self) -> Dict[str, int]:
+        return self.vocabulary
+
+    def get_all_default_fill_values(self) -> Dict[str, Scalar]:
+        return self.default_fill_values
+
+    def get_default_fill_value(self, key: str) -> Scalar:
+        return self.default_fill_values[key]
+
+    def set_default_fill_value(self, key: str, value: Scalar) -> None:
+        self.default_fill_values[key] = value
+
+    @staticmethod
+    def encoder_from_dataframe(
+        df: pd.DataFrame,
+        id_column: str,
+        target_column: str,
+        default_fill_values: Dict[str, Scalar] = DEFAULT_FILL_VALUES,
+    ) -> "TabularFeaturesInfoEncoder":
+        features_list = sorted(df.columns.values.tolist())
+        features_list.remove(id_column)
+        tab_features = TabularFeatures(
+            data=df.reset_index(), features=features_list, by=id_column, targets=target_column
+        )
+        features_to_types = tab_features.types
+        target_type = features_to_types[target_column]
+
+        # The target column is separated from the other columns
+        # so that targets and features remain separate after alignment
+        features_to_types.pop(target_column)
+
+        # extract categories information
+        ordinal_features: List[str] = sorted(tab_features.features_by_type(ORDINAL))
+        string_features: List[str] = sorted(tab_features.features_by_type(type_=STRING))
+
+        categories = {
+            ordinal_feature: sorted(df[ordinal_feature].unique().tolist()) for ordinal_feature in ordinal_features
+        }
+
+        if target_type == ORDINAL or target_type == BINARY:
+            target_categories = sorted(df[target_column].unique().tolist())
+        else:
+            target_categories = []
+
+        target_info = TargetInfoEncoder(target_column, target_type, target_categories)
+
+        # Extract vocabulary from the string columns of df
+        count_vectorizer = CountVectorizer()
```

Comment:
So it looks like we're restricting ourself to CountVectorizer() transforms for strings here. I think we can be a bit more dynamic here by exchanging information through bytes with pickle.dumps(). The poll_clients function can return a dictionary of type Dict[str, Scalar]. So bytes are an acceptable value we can pack in there. Thus, anything that is, for example, an sklearn object can be pickle.dumps(blah) and packed into those result to be materialized on the server side.